packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -3,19 +3,46 @@ import 'dart:ui';
 import 'package:flame/components.dart';
 import 'package:flame/src/palette.dart';
 
-/// Adds a collection of paints to a component
+/// Adds a collection of paints and paint layers to a component
 ///
 /// Component will always have a main Paint that can be accessed
 /// by the [paint] attribute and other paints can be manipulated/accessed
 /// using [getPaint], [setPaint] and [deletePaint] by a paintId of generic type
-/// [T], that can be omitted if the component only have one paint.
+/// [T], that can be omitted if the component only has one paint.
+/// [paintLayers] paints should be drawn in list order during the render. The
+/// main Paint is the first element.
 mixin HasPaint<T extends Object> on Component {
-  final Map<T, Paint> _paints = {};
+  late final Map<T, Paint> _paints = {};
 
-  Paint paint = BasicPalette.white.paint();
+  List<Paint>? _paintLayers;
 
-  void _assertGenerics() {
-    assert(T != Object, 'A generics type is missing on the HasPaint mixin');
+  Paint _paint = BasicPalette.white.paint();
+
+  /// List of paints to use (in order) during render.
+  List<Paint> get paintLayers {
+    if ((_paintLayers == null) || _paintLayers!.isEmpty) {
+      return _paintLayers = [_paint];
+    }
+    return _paintLayers!;
+  }
+
+  set paintLayers(List<Paint> paintLayers) {
+    _paintLayers = paintLayers;
+  }
+
+  /// Main paint. The first paint in the [paintLayers] list.
+  Paint get paint {
+    if ((_paintLayers == null) || _paintLayers!.isEmpty) {
+      return _paint;
+    }
+    return paintLayers.first;
+  }
+
+  set paint(Paint newPaint) {
+    if ((_paintLayers == null) || _paintLayers!.isEmpty) {
+      _paint = newPaint;
+    }
+    paintLayers.first = newPaint;
```

Comment:
Ha, this is a new for me

Reply:
What part?

Reply:
I am just expressing my surprise moment of realizing you can set the first element of the list by assing a value on the .frist hahah

Reply:
Ah yes, I only found this out recently too. I think it looks neater than [0] (and maybe runs a little faster without index lookup?) so went with it!

Reply:
Definitely neater! I will use it from now on.

Reply:
There's also a .last which I think is also pretty neat.